src/index.spec.ts

```diff
@@ -567,17 +577,17 @@ describe('ts-node', function () {
 
   describe('register', function () {
     let registered: tsNodeTypes.Register
+    let moduleTestPath: string
     before(() => {
       registered = register({
         project: PROJECT,
         compilerOptions: {
           jsx: 'preserve'
         }
       })
+      moduleTestPath = require.resolve('../tests/module')
     })
 
-    const moduleTestPath = require.resolve('../tests/module')
-
```

Comment:
Frankly I don't know how this was passing before this change. It should only resolve after we've registered ts-node. Maybe a race condition, where another test case already installed registered?